tabs/src/components/Leaderboard.tsx

```diff
@@ -0,0 +1,212 @@
+/// <reference path="../types/global.d.ts" />
+
+import { FunctionComponent, useEffect } from 'react';
+import styles from './Leaderboard.module.css';
+import React, { useState } from 'react';
+import {
+  getWallets,
+  getPaymentsSince,
+  getWalletBalance,
+} from '../services/lnbitsServiceLocal';
+import ZapIcon from '../images/ZapIcon.svg';
+import circleFirstPlace from '../images/circleFirstPlace.svg';
+import CircleSecondPlace from '../images/circleSecondPlace.svg';
+import circleThirdPlace from '../images/circleThirdPlace.svg';
+import circleDefaultPlace from '../images/circleDefaultPlace.svg';
+import AscendingIcon from '../images/ascending.svg';
+import DescendingIcon from '../images/descending.svg';
+
+interface FeedListProps {
+  timestamp?: number | null;
+}
+
+interface RankedWallet extends Wallet {
+  rank: number;
+  balance_msat: number;
+}
+
+const Leaderboard: React.FC<FeedListProps> = ({ timestamp }) => {
```

Comment:
We're not actually using the `timestamp` (7-days, 30-days, etc.) that is passed in.

Logic should actually be: Loop though all users in `GetUsers()`. Then for each user returned, get their private wallet `GetWalletById(User.privateWallet.id)` (note: renamed in `master` from `User.ReceivingWallet` in `global.d.ts`). Then use `GetPaymentsSince()` and add up the transactions to get a balance of received zaps.

**That said** - I think for now we could just call `getUserAccounts('balance_msat','desc')` and ignore the timestamp or that it will add the `Private` and `Allowance` wallet balances together if the above is too much effort.